src/main/scala/edu/ie3/simona/io/grid/GridProvider.scala

```diff
@@ -29,18 +32,20 @@ object GridProvider {
       case GridSourceType.CSV =>
         gridDataSource.csvParams match {
           case Some(params) =>
-            CsvGridSource
-              .readGrid(
-                simulationName,
-                params.csvSep,
-                params.directoryPath,
-                new FileNamingStrategy()
-              )
-              .getOrElse(
-                throw new InitializationException(
-                  "Error while initializing CsvGridSource! Cannot proceed without a valid GridSource!"
+            val jointGridContainer = CsvJointGridContainerSource.read(
+              simulationName,
+              params.csvSep,
+              params.directoryPath
+            )
+
+            Try.apply(ValidationUtils.check(jointGridContainer)) match {
+              case Failure(exception) =>
```

Comment:
You should also care for the case that the validation actually succeeds (which currently doesn't happen that often, I guess). If there is no accounting for the success case, we'll fail then because it's not matched.

So I'd add something like this, maybe with a more elaborate debug message:

```scala
case _ =>
  log.debug("Validation succesful")
```